src/ring_theory/jacobson_ideal.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2020 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Devon Tuma
+-/
+import ring_theory.ideal_operations
+
+universes u v
+
+namespace ideal
+variables {R : Type u} [comm_ring R]
+
+section jacobson
+
+/-- The Jacobson radical of `I` is the infimum of all maximal ideals containing `I`. -/
+def jacobson (I : ideal R) : ideal R :=
+Inf {J : ideal R | I ≤ J ∧ is_maximal J}
+
+lemma le_jacobson {I : ideal R} : I ≤ jacobson I :=
+λ x hx, mem_Inf.mpr (λ J hJ, hJ.left hx)
+
+theorem jacobson_eq_top_iff {I : ideal R} : jacobson I = ⊤ ↔ I = ⊤ :=
+⟨λ H, classical.by_contradiction $ λ hi, let ⟨M, hm, him⟩ := exists_le_maximal I hi in
+  lt_top_iff_ne_top.1 (lt_of_le_of_lt (show jacobson I ≤ M, from Inf_le ⟨him, hm⟩) $ lt_top_iff_ne_top.2 hm.1) H,
+λ H, eq_top_iff.2 $ le_Inf $ λ J ⟨hij, hj⟩, H ▸ hij⟩
+
+lemma jacobson_eq_bot {I : ideal R} : jacobson I = ⊥ → I = ⊥ :=
+λ h, eq_bot_iff.mpr (h ▸ le_jacobson)
+
+lemma jacobson.is_maximal {I : ideal R} : is_maximal I → is_maximal (jacobson I) :=
+λ h, ⟨λ htop, h.left (jacobson_eq_top_iff.1 htop),
+  λ J hJ, h.right _ (lt_of_le_of_lt le_jacobson hJ)⟩
+
+theorem mem_jacobson_iff {I : ideal R} {x : R} :
+  x ∈ jacobson I ↔ ∀ y, ∃ z, x * y * z + z - 1 ∈ I :=
+⟨λ hx y, classical.by_cases
+  (assume hxy : I ⊔ span {x * y + 1} = ⊤,
+    let ⟨p, hpi, q, hq, hpq⟩ := submodule.mem_sup.1 ((eq_top_iff_one _).1 hxy) in
+    let ⟨r, hr⟩ := mem_span_singleton.1 hq in
+    ⟨r, by rw [← one_mul r, ← mul_assoc, ← add_mul, mul_one, ← hr, ← hpq, ← neg_sub, add_sub_cancel]; exact I.neg_mem hpi⟩)
+  (assume hxy : I ⊔ span {x * y + 1} ≠ ⊤,
+    let ⟨M, hm1, hm2⟩ := exists_le_maximal _ hxy in
+    suffices x ∉ M, from (this $ mem_Inf.1 hx ⟨le_trans le_sup_left hm2, hm1⟩).elim,
+    λ hxm, hm1.1 $ (eq_top_iff_one _).2 $ add_sub_cancel' (x * y) 1 ▸ M.sub_mem
+      (le_trans le_sup_right hm2 $ mem_span_singleton.2 $ dvd_refl _)
+      (M.mul_mem_right hxm)),
+λ hx, mem_Inf.2 $ λ M ⟨him, hm⟩, classical.by_contradiction $ λ hxm,
+  let ⟨y, hy⟩ := hm.exists_inv hxm, ⟨z, hz⟩ := hx (-y) in
+  hm.1 $ (eq_top_iff_one _).2 $ sub_sub_cancel (x * -y * z + z) 1 ▸ M.sub_mem
+    (by rw [← one_mul z, ← mul_assoc, ← add_mul, mul_one, mul_neg_eq_neg_mul_symm, neg_add_eq_sub, ← neg_sub,
+        neg_mul_eq_neg_mul_symm, neg_mul_eq_mul_neg, mul_comm x y]; exact M.mul_mem_right hy)
+    (him hz)⟩
+
+@[mono] lemma jacobson_mono {I J : ideal R} : I ≤ J → I.jacobson ≤ J.jacobson :=
+begin
+  intros h x hx,
+  erw mem_Inf at ⊢ hx,
+  exact λ K ⟨hK, hK_max⟩, hx ⟨trans h hK, hK_max⟩
+end
+
+lemma comap_jacobson {S : Type v} [comm_ring S] (e : S ≃+* R) {I : ideal R}
+    : comap (e : S →+* R) (jacobson I) = jacobson (comap ↑e I) :=
```

Comment:
I've no idea about the maths here, but can `e` be replaced with either an injective or surjective function here?

Reply:
This is essentially just a special case of `comap_Inf` that already exists, with some bijectivity just simplifying the expression a bit. With the restructuring of the proofs in jacobson.lean, I don't need this anymore anyways, so I've just removed it since `comap_Inf` is essentially as powerful